consensus/view_change.go

```diff
@@ -202,10 +201,13 @@ func (consensus *Consensus) onViewChange(msg *msg_pb.Message) {
 	consensus.vcLock.Lock()
 	defer consensus.vcLock.Unlock()
 
+	// update the dictionary key if the viewID is first time received
+	consensus.addViewIDKeyIfNotExist(recvMsg.ViewID)
+
 	// TODO: remove NIL type message
 	// add self m1 or m2 type message signature and bitmap
-	_, ok1 := consensus.nilSigs[consensus.PubKey.SerializeToHexStr()]
-	_, ok2 := consensus.bhpSigs[consensus.PubKey.SerializeToHexStr()]
+	_, ok1 := consensus.nilSigs[recvMsg.ViewID][consensus.PubKey.SerializeToHexStr()]
```

Comment:
Do you need to check both the outer map returns ok and the inner map returns ok? What if there is no map associated with recvMsg.ViewID, any nil pointer issue?

Reply:
I see you have the addViewIDKeyIfNotExist, can you make sure it's called before any potential nil pointer happens?

Reply:
It is called before any access to the map